evalml/pipelines/components/transformers/encoders/ordinal_encoder.py

```diff
@@ -0,0 +1,320 @@
+                        ascending=False,
+                        kind="mergesort",
+                    )
+                    unique_values = value_counts.head(top_n).index.tolist()
+                    # Categories should be in the same order as the data's Ordinal.order categories
+                    # instead of sorted by most frequent
+                    unique_values = [
+                        cat for cat in column_ordered_categories if cat in unique_values
+                    ]
+
+                categories.append(unique_values)
+
+        # Add any null values into the categories lists so that they aren't treated as unknown values
+        # This is needed because Ordinal.order won't indicate if nulls are present, and SKOrdinalEncoder
+        # requires any null values be present in the categories list if they are to be encoded as
+        # missing values
+        if isinstance(categories, list):
+            for i, col in enumerate(X_t[self.features_to_encode]):
+                if X_t[col].isna().any():
+                    categories[i] += [np.nan]
+
+        encoded_missing_value = self.parameters["encoded_missing_value"]
+        if encoded_missing_value is None:
+            encoded_missing_value = np.nan
+
+        self._encoder = SKOrdinalEncoder(
+            categories=categories,
+            handle_unknown=self.parameters["handle_unknown"],
+            unknown_value=self.parameters["unknown_value"],
+            encoded_missing_value=encoded_missing_value,
+        )
+
+        self._encoder.fit(X_t[self.features_to_encode])
+        return self
+
+    def transform(self, X, y=None):
+        """Ordinally encode the input data.
+
+        Args:
+            X (pd.DataFrame): Features to encode.
+            y (pd.Series): Ignored.
+
+        Returns:
+            pd.DataFrame: Transformed data, where each ordinal feature has been encoded into
+            a numerical column using where of ordinal integers represent
+            the relative order of categories.
+        """
+        X = infer_feature_types(X)
+
+        X_copy = X.ww.copy()
+        X = X.ww.drop(columns=self.features_to_encode)
+
+        # Call sklearn's transform on only the ordinal columns
+        if len(self.features_to_encode) > 0:
+            X_ord = pd.DataFrame(
+                self._encoder.transform(X_copy[self.features_to_encode]),
+                index=X_copy.index,
+            )
+            X_ord.columns = self._get_feature_names()
+            X_ord.ww.init(logical_types={c: "Double" for c in X_ord.columns})
+            self._feature_names = X_ord.columns
+
+            X = ww.utils.concat_columns([X, X_ord])
+
+        return X
+
+    def _get_feature_names(self):
+        """Return feature names for the ordinal features after fitting.
+
+        Since ordinal encoding creates one encoded feature per column in features_to_encode, feature
+        names are formatted as {column_name}_ordinally_encoded
```

Comment:
Biggest nitpick: can we name them `{column_name}_ordinal_encoding`? It feels neater idk

Reply:
Of course! I'm not confident that "ordinally" is even a word, so I'm happy to change

Reply:
Changed!